File: backend/app/services/found_item_service.py
import uuid

from boto3.dynamodb.conditions import Key

from app.config import found_items_table


def create_found_item(item, current_user):
    item_dict = item.model_dump()

    item_dict["item_id"] = str(uuid.uuid4())
    item_dict["user_id"] = current_user["user_id"]

    found_items_table.put_item(Item=item_dict)

    return item_dict


def get_all_found_items():
    response = found_items_table.scan()

    return response.get("Items", [])


def get_my_found_items(current_user):
    response = found_items_table.query(
        IndexName="user_id-index",
        KeyConditionExpression=Key("user_id").eq(
            current_user["user_id"]
        ),
    )

    return response.get("Items", [])


def get_found_item(item_id: str):
    response = found_items_table.get_item(
        Key={
            "item_id": item_id,
        }
    )

    return response.get("Item")


def update_found_item(
    item_id: str,
    item,
    current_user,
):
    item_dict = item.model_dump()

    item_dict["item_id"] = item_id
    item_dict["user_id"] = current_user["user_id"]

    found_items_table.put_item(Item=item_dict)

    return item_dict


def delete_found_item(item_id: str):
    found_items_table.delete_item(
        Key={
            "item_id": item_id,
        }
    )

File: frontend/src/services/foundItemService.ts
import type {
  FoundItem,
  FoundItemRecord,
} from "../types/foundItem";
import api from "./api";

export async function createFoundItem(
  data: FoundItem
) {
  const response = await api.post(
    "/found-items/",
    data
  );

  return response.data.data;
}

export async function getFoundItems(): Promise<
  FoundItemRecord[]
> {
  const response = await api.get(
    "/found-items/"
  );

  return response.data.data;
}

export async function getMyFoundItems(): Promise<
  FoundItemRecord[]
> {
  const response = await api.get(
    "/found-items/my-items"
  );

  return response.data.data;
}

export async function getFoundItem(
  itemId: string
): Promise<FoundItemRecord> {
  const response = await api.get(
    `/found-items/${itemId}`
  );

  return response.data.data;
}

export async function updateFoundItem(
  itemId: string,
  data: FoundItem
) {
  const response = await api.put(
    `/found-items/${itemId}`,
    data
  );

  return response.data.data;
}

export async function deleteFoundItem(
  itemId: string
) {
  const response = await api.delete(
    `/found-items/${itemId}`
  );

  return response.data;
}